### DISCUSSION TITLE
ComboBox maximum width

### DISCUSSION THREAD

**Discussion by Barugon** [emilk/egui#1829] [Q&A]
Is it possible to specify a maximum width for a combo box, so that if a very long item is selected it won't expand the combo box beyond that hard limit but instead simply clip the text?

**Comment by johan-overbye** [comment]
I'm stumped on this as well. On a related note, it's unintuitive that ComboBox isn't a Widget IMO.

**Comment by yunfengwangluo** [comment]
ui.set_min_size(vec2(0.0, 152.0));
ui.set_max_size(vec2(0.0, 152.0));

**Reply by Barugon** [reply]
> ui.set_min_size(vec2(0.0, 152.0));
> ui.set_max_size(vec2(0.0, 152.0));

Thanks, I'll give that a try.

**Comment by lilly-lizard** [comment]
Found that setting the `ComboBox` width to 0 will make the width clamp to the length of the text (exactly what I wanted) e.g.
```rust
ComboBox::new("Cartesian axis", "")
    .width(0_f32)
    .selected_text(c_axis_mut.as_str())
    .show_ui(ui_h, |ui_op| {
        for (c_axis_variant, c_axis_name) in CartesianAxis::variants_with_names() {
            ui_op.selectable_value(c_axis_mut, c_axis_variant, c_axis_name);
        }
    });
```
Ends up looking like this:
![image](https://github.com/emilk/egui/assets/44878819/a1a207dd-7927-4989-b8aa-99c440512354)
